docs/search.md

```diff
@@ -1,42 +1,160 @@
+
 ### Example requests
 
 - [http://o2r.uni-muenster.de/api/v1/search?q=*](http://o2r.uni-muenster.de/api/v1/search?q=*)
-- [http://o2r.uni-muenster.de/api/v1/search?q=term](http://o2r.uni-muenster.de/api/v1/search?q=term)
-- [http://o2r.uni-muenster.de/api/v1/search?q=*&fields=compendium_id](http://o2r.uni-muenster.de/api/v1/search?q=*&fields=compendium_id)
+- [http://o2r.uni-muenster.de/api/v1/search?q=europe temperature data analysis](http://o2r.uni-muenster.de/api/v1/search?q=europe temperature data analysis)
+- [http://o2r.uni-muenster.de/api/v1/search?q=europe%20temperature%20data%20analysis](http://o2r.uni-muenster.de/api/v1/search?q=europe%20temperature%20data%20analysis)
+- [http://o2r.uni-muenster.de/api/v1/search?q=10.5555%2F12345678](http://o2r.uni-muenster.de/api/v1/search?q=10.5555%2F12345678)
+
+## Complex Search
+
+A complex search is enabled via `POST` requests with a `JSON` payload as `HTTP POST` data (_not_ `multipart/form-data`) accepting an `application/json` content type as response.
+Queries can include filters, aggregation and spatio-temporal operations as defined in the [Elasticsearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html).
 
-### Security
+`curl -X POST -H 'Content-Type: application/json' 'https://.../api/v1/search' -d '$QUERY_DSL'`
 
-Only reading requsts, i.e. `GET` requests, are allowed at the moment. This means that Kibana cannot be run against this interface, and neither can complex queries requiring `POST` requests. `GET` requests with payloads are allowed.
+The **response** structure is the same as for [simple search](#simple-search).
 
-Example query:
+### Query fields  for complex search
+
+The following fields are especially relevant to build queries.
+
+- `metadata.o2r.temporal.begin` and `metadata.o2r.temporal.end` provide a compendium's temporal extent
+- `metadata.o2r.spatial.geometry` has the compendium's spatial extent
+
+Besides these fields, all metadata of the [`o2r` metadata format](compendium/metadata.md#metadata-formats) can be used.
+
+### Examples
+
+#### Temporal search
 
 ```bash
-curl -XGET 'https://…/api/v1/search' -d '{
-    "query": {
-            "type" : {
-                "value" : "jobs"
-            }
-        }
-    }
+POST /api/v1/search -d '{
+  "query": {
+      "bool": {
+          "must": {
+              "match_all": {}
+          },
+          "filter": [
+              {
+                  "range": {
+                      "metadata.o2r.temporal.begin": {
+                          "from": "2015-03-01T00:00:00.000Z"
+                      }
+                  }
+              },
+              {
+                  "range": {
+                      "metadata.o2r.temporal.end": {
+                          "to": "2017-10-01T00:00:00.000Z"
+                      }
+                  }
+              }
+          ]
+      }
+  },
+  "from": 0,
+  "size": 10
 }'
 ```
 
-## Suggesters - WORK IN PROGRESS
+#### Spatial search
```

Comment:
The geo_shape field has to be updated in this example to match the metadata: `metadata.o2r.spatial.union.geojson.geometry`

This query worked for me: 
```
    "query": {
        "bool": {
            "must": {
                "match_all": {}
            },
            "filter": {
                "geo_shape": {
                    "metadata.o2r.spatial.union.geojson.geometry": {
                        "shape": {
                            "type": "polygon",
                            "coordinates":
                                [
                                    [
                                        [
                                            -7.294921874999999,
                                            38.54816542304656
                                        ],
                                        [
                                            34.013671875,
                                            38.54816542304656
                                        ],
                                        [
                                            34.013671875,
                                            68.6245436634471
                                        ],
                                        [
                                            -7.294921874999999,
                                            68.6245436634471
                                        ],
                                        [
                                            -7.294921874999999,
                                            38.54816542304656
                                        ]
                                    ]
                                ]
                        },
                        "relation": "within"
                    }
                }
            }
        }
    }
```

Reply:
@nuest Additionally I had some issues getting the spatial search to work:

The GeoJSON "coordinates" array in the generated compendium had a wrong structure:
- the array wasn't nested deep enough, it must be e.g. `[ [ [1,2], [2,3], [3,4], [1,2] ] ]` for a polygon
- the last point didn't equal the first point (a rectangular polygon needs 5 points to be valid)

Maybe this was because I generated the test compendia a while ago and it could be fixed already.

Reply:
Yes, I assume the coordinate generation from the extractor works now, see https://github.com/o2r-project/o2r-meta/issues?utf8=%E2%9C%93&q=is%3Aissue%20geojson%20